src/extension.ts

```diff
@@ -266,11 +266,17 @@ function stopTypeProf(state: State) {
 }
 
 function restartTypeProf() {
-  vscode.workspace.workspaceFolders?.forEach((folder) => {
-    let state = clientSessions.get(folder);
-    if (state) stopTypeProf(state);
-    startTypeProf(folder);
-  });
+	if (!vscode.workspace.workspaceFolders) return;
+
+	stopAllSessions();
+	for (const folder of vscode.workspace.workspaceFolders) {
```

Comment:
The document says:
> There is no way to stop or break a forEach() loop other than by throwing an exception.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#description.

Thus, I use for loop here.